linera-service/src/proxy/grpc.rs

```diff
@@ -491,17 +491,19 @@ where
 
         let mut certificates = vec![];
 
-        for certificate in self
-            .0
-            .storage
-            .read_certificates(hashes)
-            .await
-            .map_err(|err| Status::from_error(Box::new(err)))?
-        {
-            if grpc_message_limiter.fits::<Certificate>(certificate.clone())? {
-                certificates.push(certificate);
-            } else {
-                break;
+        'outer: for batch in hashes.chunks(100) {
+            for certificate in self
+                .0
+                .storage
+                .read_certificates(batch.to_vec())
+                .await
+                .map_err(|err| Status::from_error(Box::new(err)))?
+            {
+                if grpc_message_limiter.fits::<Certificate>(certificate.clone())? {
+                    certificates.push(certificate);
+                } else {
+                    break 'outer;
+                }
             }
         }
 
```

Comment:
<!--__GRAPHITE_HTML_TAG_START__--><i class='graphite__hidden'>[Re: line +511]<br/><br/></i>

<!--__GRAPHITE_HTML_TAG_END__-->Might be worth to log the number of certificates we're returning from here in a prometheus gauge or histogram, so we know how many certificates we're fitting in the limit<!--__GRAPHITE_HTML_TAG_START__--><span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/2663?utm_source=unchanged-line-comment">Graphite</a>.</span>

<!--__GRAPHITE_HTML_TAG_END__-->